source/common/stream_info/filter_state_impl.cc

```diff
@@ -28,11 +28,11 @@ void FilterStateImpl::setData(absl::string_view data_name, std::shared_ptr<Objec
     // cannot be overwritten by readonly data.
     const FilterStateImpl::FilterObject* current = it->second.get();
     if (current->state_type_ == FilterState::StateType::ReadOnly) {
-      throw EnvoyException("FilterState::setData<T> called twice on same ReadOnly state.");
+      IS_ENVOY_BUG("FilterState::setData<T> called twice on same ReadOnly state.");
     }
 
     if (current->state_type_ != state_type) {
-      throw EnvoyException("FilterState::setData<T> called twice with different state types.");
+      IS_ENVOY_BUG("FilterState::setData<T> called twice with different state types.");
```

Comment:
I wonder if we somehow break some invariant by letting this fall through. I can't think of any but we should be careful here.

Reply:
Yeah, I had the same thought. Maybe I should `return` after each of the `ENVOY_BUG`s, which will yield more-similar behave to the current exception path.